src/model/XRef.tsx

```diff
@@ -0,0 +1,207 @@
+  primaryAccession: string;
+};
+
+type XrefData = {
+  databaseCategory: string;
+  xref: DatabaseCrossReference;
+};
+
+type XRefProps = {
+  section: EntrySectionType;
+  data: Data;
+};
+
+type XRefItemProps = {
+  xRefEntry: DatabaseCrossReference;
+  accession: string;
+};
+
+type XRefListProps = {
+  accession: string;
+  xRefData: XrefData[];
+  database: string;
+};
+
+type XRefCategoryTableProps = {
+  databaseCategory: DatabaseCategory;
+  databases: Database[];
+  xRefData: XrefData[];
+  accession: string;
+};
+
+const databaseToDatabaseInfo: DatabaseToDatabaseInfo = databaseToDatabaseInfoJson;
+
+// Combine multiple maps into single map for quick read access.
+const entrySectionDatabaseToDatabaseCategory = new Map<
+  string,
+  DatabaseCategory
+>();
+for (const [
+  entrySection,
+  databaseCategories,
+] of entrySectionToDatabaseCategories) {
+  if (databaseCategories) {
+    for (const databaseCategory of databaseCategories) {
+      const databases = databaseCategoryToDatabases.get(databaseCategory);
+      if (databases) {
+        for (const database of databases) {
+          entrySectionDatabaseToDatabaseCategory.set(
+            `${entrySection}_${database}`,
+            databaseCategory
+          );
+        }
+      }
+    }
+  }
+}
+
+const XRefItem: React.FC<XRefItemProps> = ({ xRefEntry, accession }) => {
+  const { databaseType, properties: entryProperties, id } = xRefEntry;
+  if (
+    !id ||
+    !databaseType ||
+    !accession ||
+    !(databaseType in databaseToDatabaseInfo)
+  ) {
+    return null;
+  }
+  const info = databaseToDatabaseInfo[databaseType];
+  let uri = info.uriLink.replace(/%acc/g, accession).replace(/%value/g, id);
+  let properties: string = '';
+  if (entryProperties) {
+    properties = entryProperties
+      .map((property: Property) =>
+        property.value && property.value === '-' ? '' : property.value
+      )
+      .join(' ');
+  }
+  return (
+    <Fragment>
+      <ExternalLink url={uri}>{xRefEntry.id}</ExternalLink>
+      {properties}
+    </Fragment>
+  );
+};
+
+const XRefList: React.FC<XRefListProps> = ({
+  database,
+  xRefData,
+  accession,
+}) => {
+  const content = xRefData
+    .filter(
+      (xRefDatum: XrefData) =>
+        idx(xRefDatum, _ => _.xref.databaseType) === database
+    )
+    .map((xRefDatum: XrefData) => {
+      const { xref } = xRefDatum;
+      if (!xref) {
+        return null;
+      }
+      return (
+        <ul key={v1()}>
+          <XRefItem xRefEntry={xref} accession={accession} />
+        </ul>
+      );
+    });
+  const infoData = [{ title: database, content }];
+  return <InfoList infoData={infoData} />;
+};
+
+const XRefCategoryTable: React.FC<XRefCategoryTableProps> = ({
+  databaseCategory,
+  databases,
+  xRefData,
+  accession,
+}) => {
+  const databaseCategoryString = databaseCategoryToString.get(databaseCategory);
+  const title = databaseCategoryString && <h4>{databaseCategoryString}</h4>;
```

Comment:
`title` might be `undefined` in some cases. Better to have this as a ternary with a `null` in case `databaseCategoryString` didn't have a truthy value.

Reply:
Short-circuit evaluation seems to be a common pattern in React and keeps things concise. How about putting the evaluation in the return - would this seem more natural to you?

```
<div>
      <hr />
      {databaseCategoryString && <h4>{databaseCategoryString}</h4>}
      {databases.sort().map((database: string) => (
...
```

Reply:
Yes, that looks better and cleaner!

Reply:
Done.